BUG: Add check for array lengths in from_arrays method (GH13599)

sahildua2305 opened the pull request:
- [x] closes #13599 
- [x] whatsnew entry

sahildua2305 committed:
BUG: Add check for array lengths in from_arrays method (GH13599)

sahildua2305 committed:
BUG: Add check for array lengths in from_arrays method (GH13599)

jreback commented:
tests!

sahildua2305 committed:
BUG: Add test for array length mismatch

sahildua2305 committed:
BUG: Add test for array length mismatch

sahildua2305 committed:
BUG: Fix minor issue with new test for from_arrays

sahildua2305 committed:
BUG: Fix minor issue with new test for from_arrays

jreback commented on pandas/tests/indexes/test_multi.py:
```diff
@@ -632,6 +632,13 @@ def test_from_arrays_index_series_period(self):
 
+    def test_from_arrays_different_lengths(self):
+        # GH13599
+        idx1 = [1, 2, 3]
```
add a positive example (e.g. it doesn't raise)

sahildua2305 replied:
I am not sure if I can assert explicitly that the ValueError isn't raised.

Should I simply add an example where array lengths are same and assert the response is the MultiIndex object with valid data?

jreback replied:
you simply construct an expected MultiIndex and use ``.equals`` that it is equal to a result that uses ``.from_arrays``. I am sure there are other examples to model from.

sahildua2305 replied:
@jreback is that test really required? The same input array length case is anyway getting checked in all other `from_arrays` related tests.

jreback replied:
better yet make sure to check with a 0-len array (check for all 2 cases, idx1 or idx2 is 0-len)

jreback commented:
pls add a whatsnew note for 0.19.0

yui-knk committed:
CLN: Fix compile time warnings

This commit suppresses these warnings

warning: comparison of constant -1 with expression\
of type 'PANDAS_DATETIMEUNIT' is always true\
[-Wtautological-constant-out-of-range-compare]

Author: yui-knk <spiketeika@gmail.com>

Closes #13607 from yui-knk/fix_c_warning and squashes the following commits:

e9eee1d [yui-knk] CLN: Fix compile time warnings

jorisvandenbossche committed:
Pin IPython for doc build to 4.x (see #13639)

jreback committed:
CLN: reorg type inference & introspection

closes #12503

Author: Jeff Reback <jeff@reback.net>

Closes #13147 from jreback/types and squashes the following commits:

244649a [Jeff Reback] CLN: reorg type inference & introspection

gfyoung committed:
BLD: included pandas.api.* in setup.py (#13640)

sahildua2305 committed:
Minor fix for linter

sahildua2305 committed:
Minor fix for linter

sahildua2305 committed:
Update whatsnew entry

sahildua2305 committed:
Update whatsnew entry

jorisvandenbossche committed:
DOC/BLD: pin IPython version to 4.2.0 (#13639) (#13647)

sinhrks committed:
TST: reorganize tools.tests (#13619)

yarikoptic committed:
BF(TST): allow AttributeError being raised (in addition to TypeError) from mpl (#13641)

Closes #13570

gfyoung committed:
API, DEPR: Raise and Deprecate Reshape for Pandas Objects

Author: gfyoung <gfyoung17@gmail.com>

Closes #13012 from gfyoung/categorical-reshape-validate and squashes the following commits:

3ad161d [gfyoung] API: Prevent invalid arguments to Categorical.reshape

jreback commented on doc/source/whatsnew/v0.19.0.txt:
```diff
@@ -550,3 +550,5 @@ Bug Fixes
 
 - Bug where ``pd.read_gbq()`` could throw ``ImportError: No module named discovery`` as a result of a naming conflict with another python package called apiclient  (:issue:`13454`)
+
+-Bug in ``MultiIndex.from_arrays`` doesn't check for arrays lengths (:issue:`13599`)
```
doesn't -> did not

array lengths -> input array lengths

yui-knk committed:
CLN: Fix compile time warnings

Author: yui-knk <spiketeika@gmail.com>

Closes #13643 from yui-knk/warning2 and squashes the following commits:

ee3a4fb [yui-knk] CLN: Fix compile time warnings

jorisvandenbossche committed:
CLN: fix some issues in asv benchmark suite (#13630)

* CLN: fix params list

* Fix issue in asv.conf.json for win32+other environment

Fix mistaken exclusion of virtualenv or existing:same on win32 in the config.

Credits: @pv

* CLN: remove DataMatrix

* ASV: fix exlusion of tables package for non-conda environments

sinhrks committed:
TST: add tests for Timestamp.toordinal/fromordinal

follow-up for #13593

Author: sinhrks <sinhrks@gmail.com>

Closes #13610 from sinhrks/depr_timestamp_offset2 and squashes the following commits:

28f8d41 [sinhrks] TST: add tests for Timestamp.toordinal

toobaz committed:
CLN: Initialization coincides with mapping, hence with uniqueness check

 - [x] tests added / passed   - [x] passes ``git diff upstream/master
| flake8 --diff``    Rebased version of
https://github.com/pydata/pandas/pull/10229 which was [actually not](h
ttps://github.com/pydata/pandas/pull/10229#issuecomment-131470116)
fixed by https://github.com/pydata/pandas/pull/10199.    Nothing
particular relevant, just wanted to delete this branch locally and
noticed it still applies: you'll judge what to do of it.

Author: Pietro Battiston <me@pietrobattiston.it>

Closes #13594 from toobaz/fix_checkunique and squashes the following commits:

a63bd12 [Pietro Battiston] CLN: Initialization coincides with mapping, hence with uniqueness check

sinhrks committed:
API: Change Period('NAT') to return NaT

closes #12759
closes #13582

Author: sinhrks <sinhrks@gmail.com>

Closes #13609 from sinhrks/period_nat and squashes the following commits:

9305c36 [sinhrks] COMPAT: Period(NaT) now returns pd.NaT

jreback committed:
BUG: construction of Series with integers on windows not default to int64

closes #13646

Author: Jeff Reback <jeff@reback.net>

Closes #13661 from jreback/foo and squashes the following commits:

e26f9bf [Jeff Reback] BUG: construction of Series with integers on windows not defaulting to int64

gfyoung committed:
CLN: Removed levels attribute from Categorical

Deprecated back in `0.15.0` and therefore long overdue.  Closes #8376.

Author: gfyoung <gfyoung17@gmail.com>

Closes #13612 from gfyoung/categorical-levels-remove and squashes the following commits:

f1254df [gfyoung] MAINT: Relocated backwards compat categorical pickle tests
f3321cb [gfyoung] CLN: Removed levels attribute from Categorical

sahildua2305 committed:
Fix minor typo

sahildua2305 committed:
Fix minor typo

shawnheide committed:
DOC: Add reference of DataFrame.rename_axis and Series.rename_axis to api.rst (#13678)

shawnheide committed:
DOC: correct template for .cum* descriptions (#13683)

Closes #13682

yui-knk committed:
DOC: fix a keyword coerce in  array_to_timedelta64 (#13686)

sinhrks committed:
TST: assert message shows unnecessary diff (#13676)

sinhrks committed:
ENH: Series.append now has ignore_index kw

Author: sinhrks <sinhrks@gmail.com>

Closes #13677 from sinhrks/append_series and squashes the following commits:

4bc7b54 [sinhrks] ENH: Series.append now has ignore_index kw

wcwagner committed:
BUG: Add  type check for width parameter in str.pad method GH13598

closes #13598

Author: wcwagner <wcw13@my.fsu.edu>

Closes #13690 from wcwagner/bug/13598 and squashes the following commits:

9669f3f [wcwagner] BUG: "Replaced isinstance with is_integer, and changed test_pad_width to use getattr"
40a3188 [wcwagner] BUG: "Switched to single test method asserting functions that use pad raise correctly."
06795db [wcwagner] BUG: "Added tests for width parameter on center, ljust, rjust, zfill."
468df3a [wcwagner] BUG: Add  type check for width parameter in str.pad method GH13598

yui-knk committed:
BUG: Cast a key to NaT before get loc from Index

closes #13603

Author: yui-knk <spiketeika@gmail.com>

Closes #13687 from yui-knk/fix_13603 and squashes the following commits:

0960395 [yui-knk] BUG: Cast a key to NaT before get loc from Index

jreback committed:
BUG: merge_asof not handling allow_exact_matches and tolerance on first entry

closes #13695

Author: Jeff Reback <jeff@reback.net>

Closes #13698 from jreback/merge_asof and squashes the following commits:

c46dcfa [Jeff Reback] BUG: merge_asof not handling allow_exact_matches and tolerance on first entry

gfyoung committed:
CLN: removed pandas.sandbox

xref #9615

Author: gfyoung <gfyoung17@gmail.com>

Closes #13670 from gfyoung/sandbox-removal and squashes the following commits:

2a014aa [gfyoung] CLN: removed pandas.sandbox

sinhrks committed:
DEPR: Remove legacy offsets

Follow-up of #10951.
Remove legacy offsets deprecated in 0.17.0.

Author: sinhrks <sinhrks@gmail.com>

Closes #13590 from sinhrks/depr_legacy_offset and squashes the following commits:

2593b1f [sinhrks] DEPR: Remove legacy offsets

gfyoung committed:
CLN: removed setter method of categorical's ordered attribute

xref #9611

Author: gfyoung <gfyoung17@gmail.com>

Closes #13671 from gfyoung/cat-set-order-removal and squashes the following commits:

58938e7 [gfyoung] CLN: removed setter method of categorical's ordered attribute

pijucha committed:
BUG/PERF: Sort mixed-int in Py3, fix Index.difference

fixes some issues from #13432
closes #12044
closes #12814

Author: Piotr Jucha <pi.jucha@gmail.com>

Closes #13514 from pijucha/setop13432 and squashes the following commits:

3a96089 [Piotr Jucha] BUG/PERF: Sort mixed-int in Py3, fix Index.difference

bashtage committed:
ENH: Add support for writing variable labels to Stata files

closes #13536
closes #13535

Author: Kevin Sheppard <kevin.k.sheppard@gmail.com>

Closes #13631 from bashtage/stata-data-labels and squashes the following commits:

1e1e1bf [Kevin Sheppard] ENH: Add support for writing variable labels

sinhrks committed:
API: Index doesn't results in PeriodIndex if Period contains NaT

Author: sinhrks <sinhrks@gmail.com>

Closes #13664 from sinhrks/period_infer2 and squashes the following commits:

b208a9e [sinhrks] API: Index doesn't results in PeriodIndex if Period contains NaT

sinhrks committed:
PERF: improve DTI string parse

closes #11169
closes #11287

Author: sinhrks <sinhrks@gmail.com>

Closes #13692 from sinhrks/dti_perf and squashes the following commits:

8774772 [sinhrks] PERF: improve DTI string parse

chris-b1 committed:
DOC: resample warnings

closes #13618
closes #13520

Author: Chris <cbartak@gmail.com>

Closes #13675 from chris-b1/resample-warning and squashes the following commits:

2185c1f [Chris] whatsnew note
c58c70c [Chris] DOC: resample warnings

gfyoung committed:
CLN: Removed the flavor='mysql' option and deprecate flavor in DataFrame.to_sql (#13611)

jreback committed:
ENH: add time-window capability to .rolling

xref #13327
closes #936

Author: Jeff Reback <jeff@reback.net>

Closes #13513 from jreback/rolling and squashes the following commits:

d8f3d73 [Jeff Reback] ENH: add time-window capability to .rolling

yui-knk committed:
CLN: Remove a test case about Timestamp to TestTimestamp (#13722)

sinhrks committed:
DOC/DEPR: pivot_annual

closes #736

Author: sinhrks <sinhrks@gmail.com>

Closes #13706 from sinhrks/pivot_annual and squashes the following commits:

d097bab [sinhrks] DOC/DEPR: pivot_annual

sinhrks committed:
PERF: Improve Period hashing

closes #12817

Author: sinhrks <sinhrks@gmail.com>

Closes #13705 from sinhrks/period_hash and squashes the following commits:

e1fb7f4 [sinhrks] PERF: Improve Period hasing

gfyoung committed:
MAINT: Removed some warnings in tests

Per discussion with @jreback <a href="https://github.com/pydata/pandas
/pull/13671#issuecomment-233508468">here</a>.

Author: gfyoung <gfyoung17@gmail.com>

Closes #13702 from gfyoung/test-warnings-remove and squashes the following commits:

e7292d3 [gfyoung] MAINT: Removed some warnings in tests

gfyoung committed:
CLN: removed the 'diff' method for Index

Deprecated all the way back in `0.15.0` <a
href="https://github.com/pydata/pandas/pull/8227">here</a>.

Author: gfyoung <gfyoung17@gmail.com>

Closes #13669 from gfyoung/remove-index-diff and squashes the following commits:

7dca659 [gfyoung] CLN: removed the 'diff' method for Index

sahildua2305 committed:
Remove minor typo

jorisvandenbossche commented:
Superceded by https://github.com/pydata/pandas/pull/13728
